### PR TITLE
add a test case for issue #32031

### PR DESCRIPTION
**Pull request by dsprenkels**
I propose a test case to finish the fix for issue #32031. Please review this commit thoroughly, as I have never written a codegen test before.

r? @eddyb

### CODE REVIEW

**Comment by eddyb** [comment]
@dsprenkels Can you add the email address you used in the commit to your github account? Otherwise the commits are not linked to your account. This is of course not necessary, let me know if it's intentional.

File: src/test/codegen/issue-32031.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// except according to those terms.
+
+// compile-flags: -C no-prepopulate-passes
+
+#![crate_type = "lib"]
+
+#[no_mangle]
+pub struct F32(f32);
+
+// CHECK: @add_newtype_f32(float, float)
```
Comment (eddyb) [inline src/test/codegen/issue-32031.rs]: Any chance this could also check the return type?

**Comment by dsprenkels** [comment]
Yup, this was indeed not intentional. Thanks for the heads-up!
(I did this commit on my server, where compilation is a lot faster.)

File: src/test/codegen/issue-32031.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#[no_mangle]
+pub fn add_newtype_f32(a: F32, b: F32) -> F32 {
+    F32(a.0 + b.0)
+}
+
+#[no_mangle]
+pub struct F64(f64);
+
+// CHECK: @add_newtype_f64(double, double)
+// CHECK: ret float
```
Comment (eddyb) [inline src/test/codegen/issue-32031.rs]: Is this not `double`?

File: src/test/codegen/issue-32031.rs (hunk 1 above)
Comment (eddyb) [inline src/test/codegen/issue-32031.rs]: What I meant, btw, is that the full line is `define float @add_newtype_f32(float, float)` - and if you include all of it, you also ensure the return type is correct.

Comment (dsprenkels) [inline src/test/codegen/issue-32031.rs]: Oh, check. I'll fix this accordingly.

**Pull request merged**